bioptim/limits/penalty_option.py

```diff
@@ -283,8 +285,11 @@ def _set_penalty_function(self, all_pn: Union[PenaltyNodeList, list, tuple], fcn
             nlp = all_pn[0].nlp
             nlp_post = all_pn[1].nlp
             name = self.name.replace("->", "_").replace(" ", "_")
-            state_cx = horzcat(nlp.states.cx_end, nlp_post.states.cx)
-            control_cx = horzcat(nlp.controls.cx_end, nlp_post.controls.cx)
+            state_cx = horzcat(vertcat(*[nlp.states.cx[i] for i in self.states_pre_idx]),
```

Comment:
What if `self.states_pre_idx` is undefined (i.e. None)? Is this automatically created later? (I did not have a look yet to later code)